### PR TITLE
raftstore: support limit the count of the peer stats reported in heartbeat (#10247)

### PR DESCRIPTION
**Pull request by nolouch**
cherry-pick #10247 to release-5.1
## Problem Summary:

The peer stats reported in one heartbeat is no limited now.

## What is changed and how it works?
Use TopN to limit the count of the reported peer stat in single store heartbeat.

### Check List
- Tests
- Unit test

## Release note
```release-note
limit the hotspot report count
```

### CODE REVIEW

**Review by NingLin-P — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Comment by Mini256** [comment]
@nolouch, Hi, for removing the `do-not-merge/release-note-label-needed` label, You need to specify the language of the code block as 'release-note'.

like this: 

    ```release-note
    None
    ```

not like:

    ```
    None
    ```

File: components/tikv_util/src/topn.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Reverse;
+use std::collections::{binary_heap, BinaryHeap};
+use std::iter;
+
+// todo: Make it TopN<T, const N: usize> when we update our Rust version
+/// TopN is used to collect the largest `cap` items pushed in
```
Comment (nida624) [inline components/tikv_util/src/topn.rs:8]: components/tikv_util/src/topn.rs

Comment (nida624) [reply components/tikv_util/src/topn.rs:8]: components/tikv_util/src/topn.rs

File: tests/integrations/raftstore/test_stats.rs [hunk 2]
```diff
@@ -101,33 +101,46 @@ fn test_node_simple_store_stats() {
 fn test_store_heartbeat_report_hotspots() {
     fail::cfg("mock_hotspot_threshold", "return(0)").unwrap();
     fail::cfg("mock_tick_interval", "return(0)").unwrap();
-    let (cluster, client, ctx) = must_new_and_configure_cluster_and_kv_client(|cluster| {
```
Comment (nida624) [inline tests/integrations/raftstore/test_stats.rs:104]: components/tikv_util/src/topn.rs

File: components/tikv_util/src/lib.rs [hunk 3]
```diff
@@ -41,6 +41,7 @@ pub mod sys;
 pub mod thread_group;
```
Comment (nida624) [inline components/tikv_util/src/lib.rs:41]: components/tikv_util/src/topn.rs

**Review by nida624 — COMMENTED** [review]
components/tikv_util/src/topn.rs

**Pull request merged**